### ISSUE TITLE
Underworld Mission (Campaign) 06: "Dark Ally" Empire keeps retreating.

### ISSUE THREAD

**Issue by AlyMar1994** [AlyMar1994/UFoCP#14]
In Underworld Campaign #06, "Dark Ally" (the mission of saving Silri from Dathomir), when outnumbered enough, it's possible the Empire may retreat before the mission is fully completed, preventing the ending of the actual mission.

Nothing is truly affected by this, only that the story progression is somewhat hampered and it looks extremely unprofessional that the mission can just end even when it isn't realistic to expect the Empire to retreat.

**Comment by AlyMar1994** [comment]
Haven't done even remotely much with campaign/story editing, though I guess now would be a good time to start.

NOTE: Through some method it's probably possible to apply a Grand Moff Tarkin-like effect to the units of the mission preventing retreat.

**Comment by KadVenku** [comment]
You can turn off retreat by faction with an xml story event object.
There is one for the Consortium already, so you could turn that off for the Empire too:

```xml
	<Event Name="Underworld_A02M06_Prevent_Retreat">
		<Event_Type>STORY_TRIGGER</Event_Type>
		<Reward_Type>DISABLE_RETREAT</Reward_Type>
		<Reward_Param1>UNDERWORLD</Reward_Param1>
		<Reward_Param2>1</Reward_Param2>
		<Prereq>Underworld_A02M06_Begin</Prereq>
	</Event>
```

Possible side effects are that the AI tries to retreat, but can't. But that would have to be tested.

**Comment by AlyMar1994** [comment]
Oh Jesus-Tap-Dancing-Christ! I didn't know that affected the AI aswell.

I found the story event a long while ago, I found some (old) tutorials about them figured out how to interpret the files, but never knew these effects could apply that to the Empire, I thought it was player-only. When I get home later today, I'll implement the change into v1.5's branch and test Dark Ally.

Thank you Venku, a huge help.

**Comment by AlyMar1994** [comment]
The fix's been implemented, and is an automatic procedure, along with an undo for the end of the mission. For all intents and purposes, this issue is fixed in an upcoming update for v1.5.

Added entries:


	<Event Name="Empire_A02M06_Prevent_Retreat">
		<Event_Type>STORY_TRIGGER</Event_Type>
		<Reward_Type>DISABLE_RETREAT</Reward_Type>
		<Reward_Param1>EMPIRE</Reward_Param1>
		<Reward_Param2>1</Reward_Param2>
		<Prereq>Underworld_A02M06_Begin</Prereq>
	</Event>



	<Event Name="Empire_A02M06_Re-Enable_Retreat">
		<Event_Type>STORY_TRIGGER</Event_Type>
		<Reward_Type>DISABLE_RETREAT</Reward_Type>
		<Reward_Param1>EMPIRE</Reward_Param1>
		<Reward_Param2>0</Reward_Param2>
		<Prereq>Underworld_A02M06_Governors_Destroyed</Prereq>
	</Event>
